ante/fee.go

```diff
@@ -89,23 +92,40 @@ func IsOracleOrGravityTx(msgs []sdk.Msg) bool {
 
 // getTxPriority returns naive tx priority based on the lowest fee amount (regardless of the
 // denom) and oracle tx check.
-func getTxPriority(fee sdk.Coins, isOracleOrGravity bool) int64 {
+func getTxPriority(fees sdk.Coins, gasAmount int64, isOracleOrGravity bool, msgs []sdk.Msg) int64 { //nolint: unparam
 	var priority int64
-	for _, c := range fee {
-		// TODO: we should better compare amounts
-		// https://github.com/umee-network/umee/issues/510
+	/* TODO: IBC tx prioritization is not stable and we will implement a more general
+	 * tx prioritization once that will be resolved
+	 * https://github.com/umee-network/umee/issues/1289
+	for _, c := range fees {
 		p := int64(math.MaxInt64)
-		if c.Amount.IsInt64() {
-			p = c.Amount.Int64()
+		gasPrice := c.Amount.QuoRaw(gasAmount)
+		if gasPrice.IsInt64() {
+			p = gasPrice.Int64()
 		}
 		if priority == 0 || p < priority {
 			priority = p
 		}
 	}
+	*/
 	if isOracleOrGravity {
-		// TODO: this is a naive version.
-		// Proper solution will be implemented in https://github.com/umee-network/umee/issues/510
-		priority += 100000
+		return 100
 	}
+	for _, msg := range msgs {
```

Comment:
This seems a bit weird because `p` will have the least priority of all of the messages  :thinking:

Reply:
It's desirable. You should not add spam / rubbish messages to your transaction just to have extra priority.